src/main/java/org/folio/ed/controller/ApiKeyService.java

```diff
@@ -0,0 +1,75 @@
+package org.folio.ed.controller;
+
+import static org.folio.edge.core.Constants.HEADER_API_KEY;
+import static org.folio.edge.core.Constants.PARAM_API_KEY;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.ServletRequest;
+
+import org.apache.catalina.connector.RequestFacade;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ApiKeyService {
+
+  @Value("${api_key_sources}")
+  private String apiKeySources;
+
+  public static final Pattern AUTH_TYPE = Pattern.compile("(?i).*apikey (\\w*).*");
+  public static final Pattern COMMA = Pattern.compile(",");
+
+  private List<ApiKeySource> sources;
+
+  @PostConstruct
+  public void init() {
+    sources = new ArrayList<>();
+    for (String source : COMMA.split(apiKeySources)) {
+      sources.add(ApiKeySource.valueOf(source));
+    }
+  }
+
+  public String getEdgeApiKey(ServletRequest servletRequest) {
+    for (ApiKeySource source : sources) {
+      String apiKey = null;
+      if (ApiKeySource.PARAM == source ) {
+        apiKey = getFromParam(servletRequest);
+      } else if (ApiKeySource.HEADER == source) {
+        apiKey = getFromHeader(servletRequest);
```

Comment:
(minor) Can we simplify this by not using additional variable like apiKey and just return getFromParam(servletRequest) or getFromHeader(servletRequest) or null ? Also we can use map or strategy here to get rid from multiple if statements